SourceCode/Homework 2/task2.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <fstream>
+#include <cmath>
+
+const int imageWidth = 1920;
+const int imageHeight = 1080;
+const int maxColorComponent = 255;
+
+void generatePPM(const std::string& filename) {
+    std::ofstream ppmFileStream(filename);
+    if (!ppmFileStream) {
+        std::cerr << "Error opening file." << std::endl;
+        return;
+    }
+
+    ppmFileStream << "P3\n" << imageWidth << " " << imageHeight << "\n" << maxColorComponent << "\n";
+
+    int centerX = imageWidth / 2;
+    int centerY = imageHeight / 2;
+    int halfWidth = 300;
+    int halfHeight = 400;

```

Comment:
Seems like these values could be calculated dynamically based on the image dimensions.